bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSHandler.java

```diff
@@ -334,4 +355,52 @@ public void postUpdate(String channelId, State state) {
     public Collection<Class<? extends ThingHandlerService>> getServices() {
         return Collections.singleton(LGWebOSActions.class);
     }
+
+    @Override
+    public void thingDiscovered(DiscoveryService source, DiscoveryResult result) {
+        if (THING_TYPE_WEBOSTV.equals(result.getThingTypeUID())) {
+            Map<String, Object> properties = result.getProperties();
+            String host = (String) properties.get(CONFIG_HOST);
+            if (host != null && host.equals(getLGWebOSConfig().getHost())) {
+                // Thing matching the discovery
+                logger.debug("thingDiscovered: discovery with UID {} matching thing {}", result.getThingUID(),
+                        getThing().getUID());
+
+                channelHandlers.get(CHANNEL_POWER).onDeviceReady(CHANNEL_POWER, this);
+
+                // Update the thing properties from the discovery result
+                Map<String, String> map = new HashMap<>();
+                String deviceId = (String) properties.get(PROPERTY_DEVICE_ID);
+                if (deviceId != null && !deviceId.isEmpty()) {
+                    udn = deviceId;
+                    logger.debug("UDN {} considered", udn);
+                    map.put(PROPERTY_DEVICE_ID, deviceId);
+                }
+                String manufacturer = (String) properties.get(PROPERTY_MANUFACTURER);
+                if (manufacturer != null && !manufacturer.isEmpty()) {
+                    map.put(PROPERTY_MANUFACTURER, manufacturer);
+                }
+                String modelName = (String) properties.get(PROPERTY_MODEL_NAME);
+                if (modelName != null && !modelName.isEmpty()) {
+                    map.put(PROPERTY_MODEL_NAME, modelName);
+                }
+                storeProperties(map);
+            }
+        }
+    }
+
+    @Override
+    public void thingRemoved(DiscoveryService source, ThingUID thingUID) {
+        if (!udn.isEmpty() && thingUID.getId().equals(udn)) {
+            logger.debug("thingRemoved: discovery with UID {} matching thing {}", thingUID, getThing().getUID());
+            channelHandlers.get(CHANNEL_POWER).onDeviceRemoved(CHANNEL_POWER, this);
```

Comment:
same comment as in thing discovered.

Reply:
This line is the most important of this PR. It does what you expect since a long time, that is updating the power channel to OFF as fast as possible, even when the websocket is still connected.

Reply:
Considering the PR #7355, I think we should trigger the new state DISCONNECTING in this case.